### PR TITLE
fix: json module parse error

### LINKED ISSUE #2757: [Bug]: Gracefully handle JSON module parse errors
### Reproduction link or steps

https://stackblitz.com/edit/github-csznuh?file=src%2Fdata.json

### What is expected?

```sh
[PARSE_ERROR] Error: Unexpected `,`
    ╭─[src/data.json:2:22]
    │
  2 │   "trailing comma": "",
    │                       ┬
    │                       ╰── unexpected
────╯
```

### What is actually happening?

```sh
[UNHANDLEABLE_ERROR] Error: Something wrong inside the rolldown, please report this at https://github.com/rolldown/rolldown/issues.
trailing comma at line 3 column 1
```

### System Info

```Shell
Rolldown 0.14.0
```

### Any additional comments?

I confuse JS objects for JSON sometimes...

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Closed #2757

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
* **#2761** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2761?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2760** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2760?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2760?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown_error/src/build_error/error_constructors.rs [hunk 1]
```diff
@@ -261,6 +263,19 @@ impl BuildDiagnostic {
...
 
+  #[allow(clippy::cast_possible_truncation)]
+  pub fn json_parse(
+    filename: ArcStr,
+    source: ArcStr,
+    line: usize,
+    column: usize,
+    message: ArcStr,
+  ) -> Self {
+    let start_offset = line_column_to_byte_offset(source.as_str(), line - 1, column - 1);
```
Comment (IWANABETHATGUY) [inline crates/rolldown_error/src/build_error/error_constructors.rs:275]: serde_json err is one-based https://docs.rs/serde_json/1.0.132/serde_json/struct.Error.html#method.column

Comment (hyf0) [reply crates/rolldown_error/src/build_error/error_constructors.rs:275]: Would it be better to comment in code directly?

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown_error/src/build_error/error_constructors.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/rolldown_error/src/build_error/error_constructors.rs:275]: ok

**Pull request merged**